From 01850e0f517db459d1fd10d18b72efd673c23006 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 14 Sep 2012 13:00:57 +0200
Subject: [PATCH] Override the Pod::Simple::parse_file

This sets output_fh to STDOUT if it's not already set.
This resolves CPANRT#77530 and RHBZ#826872 and is fixed in podlators-2.4.1.
---
 lib/Pod/Man.pm  | 11 +++++++++++
 lib/Pod/Text.pm | 11 +++++++++++
 2 files changed, 22 insertions(+)

diff --git a/lib/Pod/Man.pm b/lib/Pod/Man.pm
index 96f3fcc..ad5e5ac 100644
--- a/lib/Pod/Man.pm
+++ b/lib/Pod/Man.pm
@@ -1302,6 +1302,17 @@ sub parse_from_filehandle {
     $self->parse_from_file (@_);
 }
 
+# Pod::Simple's parse_file doesn't set output_fh.  Wrap the call and do so
+# ourself unless it was already set by the caller, since our documentation has
+# always said that this should work.
+sub parse_file {
+    my ($self, $in) = @_;
+    unless (defined $$self{output_fh}) {
+        $self->output_fh (\*STDOUT);
+    }
+    return $self->SUPER::parse_file ($in);
+}
+
 ##############################################################################
 # Translation tables
 ##############################################################################
diff --git a/lib/Pod/Text.pm b/lib/Pod/Text.pm
index cc02820..1a8b0bf 100644
--- a/lib/Pod/Text.pm
+++ b/lib/Pod/Text.pm
@@ -679,6 +679,17 @@ sub parse_from_filehandle {
     $self->parse_from_file (@_);
 }
 
+# Pod::Simple's parse_file doesn't set output_fh.  Wrap the call and do so
+# ourself unless it was already set by the caller, since our documentation has
+# always said that this should work.
+sub parse_file {
+    my ($self, $in) = @_;
+    unless (defined $$self{output_fh}) {
+        $self->output_fh (\*STDOUT);
+    }
+    return $self->SUPER::parse_file ($in);
+}
+
 ##############################################################################
 # Module return value and documentation
 ##############################################################################
-- 
1.7.11.4

